src/intrinsics/SetPrototype.mts

```diff
@@ -355,3 +709,38 @@ function SetDataHas(resultSetData: (Value | undefined)[], value: Value) {
   // 2. Return false.
   return Value.false;
 }
```

Comment:
I suggest change `SetDataHas` to return a `boolean` instead of a `BooleanValue` since it's return value is never used as a `Value`.